File: backend/src/main/java/com/handmade/store/service/CartService.java
package com.handmade.store.service;

import com.handmade.store.dto.cart.CartRequest;
import com.handmade.store.dto.cart.CartResponse;

import java.util.Map;

public interface CartService {

    CartResponse getCart(String email);

    CartResponse addToCart(String email, CartRequest request);

    CartResponse updateCartItem(String email, Long productId, int quantity);

    Map<String, Object> removeFromCart(String email, Long productId);

    Map<String, Object> clearCart(String email);
}

File: frontend/src/services/cartService.js
import api from './api';

export const getCart = () => {
  return api.get('/cart');
};

export const addToCart = (data) => {
  return api.post('/cart', data);
};

export const updateCartItem = (productId, quantity) => {
  return api.put(`/cart/${productId}`, null, { params: { quantity } });
};

export const removeFromCart = (productId) => {
  return api.delete(`/cart/${productId}`);
};

export const clearCart = () => {
  return api.delete('/cart');
};
